src/splitting.rs

```diff
@@ -41,6 +41,21 @@ pub trait WordSplitter: std::fmt::Debug {
     fn split<'w>(&self, word: &'w str) -> Vec<(&'w str, &'w str, &'w str)>;
 }
 
+impl WordSplitter for Box<dyn WordSplitter> {
+    fn split<'w>(&self, word: &'w str) -> Vec<(&'w str, &'w str, &'w str)> {
+        use std::ops::Deref;
+        self.deref().split(word)
+    }
+}
+/* Alternative, also adds impls for specific Box<S> i.e. Box<HyphenSplitter>
```

Comment:
Ah, this was the example you mentioned in your PR description. I'm not sure exactly what the pros/cons are of these two options.

Reply:
I'll be happy to simply leave this out for now and then merge the rest — then we can make forward progress.